compat-win32.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#ifdef _WIN32
+
+/* Redefine _WIN32_WINNT */
+#ifdef _WIN32_WINNT
+#undef _WIN32_WINNT
+#endif
+#define _WIN32_WINNT 0x0600
```

Comment:
why is this necessary?

Reply:
This macro defines Windows system version, `0x0600` for Windwos Vista . Case `AI_ADDRCONFIG` was used in `tcp_connect` and defined in `ws2tcpip.h`, minimum value of `_WIN32_WINNT` should be `0x0600`, which is defined to `0x502` in `_mingw.h` by default.

Reply:
this should probably be handled by the `configure` script.

Just like it defines `_DARWIN_C_SOURCE` for Darwin / Mac OS X, it should define `_WIN32_WINNT` for Windows.
E.g.:
``` autoconf
case "$host_os" in
...
*-mingw32)
    AC_DEFINE([_WIN32_WINNT], [0x0600], [Define on Windows to expose AI_ADDRCONFIG in WinSock2])
;;
```

This is specially important because the definition is conditional, and ultimately depends on the software to request the right API.